### PR TITLE
Fix processing empty SQL and GraphQL files, refactor codegen

### PR DESCRIPTION
**Pull request by droserasprout**

### CODE REVIEW

**Comment by droserasprout** [comment]
[Ignore empty sql scripts](https://trello.com/c/YiDCiX5x/197-ignore-empty-sql-scripts)

**Comment by droserasprout** [comment]
[Generate types for templates used in dynamic indexes](https://trello.com/c/lCnsU4A4/150-generate-types-for-templates-used-in-dynamic-indexes)

File: src/dipdup/dipdup.py [hunk 1]
```diff
@@ -336,14 +336,9 @@ async def _execute_sql_scripts(self, reindex: bool) -> None:
...
-                continue
-
-            with open(join(sql_path, filename)) as file:
-                sql = file.read()
-
-            self._logger.info('Executing `%s`', filename)
+        for file in iter_files(sql_path, '.sql'):
+            self._logger.info('Executing `%s`', file.name)
+            sql = file.read()
             await get_connection(None).execute_script(sql)
```
Comment (m-kus) [inline src/dipdup/dipdup.py:342]: Typical issue: commented content thus the file is not empty but the script is invalid

File: src/dipdup/codegen.py [hunk 2]
```diff
@@ -121,13 +123,30 @@ async def create_package(self) -> None:
...
-        for index_config in self._config.indexes.values():
+        index_configs: List[IndexConfigT]
+        if template:
+            if not contract:
+                raise RuntimeError('Both `template` and `contract` arguments are required')
+            # FIXME: Implement classmethods to avoid adding temporary index
+            self._config.indexes['TEMP'] = IndexTemplateConfig(
+                template=template,
+                # NOTE: Regex magic! Replace all variables. What could possibly go wrong?
+                values={'\w*': contract},
```
Comment (m-kus) [inline src/dipdup/codegen.py:140]: Template entrypoints, template callback names, etc :)

File: src/dipdup/dipdup.py (hunk 1 above)
Comment (droserasprout) [reply src/dipdup/dipdup.py:342]: Check for "de facto empty" files would be too complex — GraphQL has multiline comments. Wrapped exception thrown on empty SQL.

**Review by m-kus — APPROVED** [review]

**Pull request merged**